tests/test_optimizer.py

```diff
@@ -370,3 +373,52 @@ def test_maxiter_1_and_0_solve():
             maxiter=0, constraints=constraints, objective=obj, optimizer=opt
         )
         assert result["nfev"] <= 1
+
+
+@pytest.mark.unit
+@pytest.mark.slow
+def test_scipy_fail_message():
+    """Test that scipy fail message does not cause an error (see PR #434)."""
+    constraints = (
+        FixBoundaryR(fixed_boundary=True),
+        FixBoundaryZ(fixed_boundary=True),
+        FixPressure(),
+        FixCurrent(),
+        FixPsi(),
+    )
+    objectives = ForceBalance()
+    obj = ObjectiveFunction(objectives)
+    eq = Equilibrium()
+    # should fail on maxiter, and should NOT throw an error
+    for opt in ["scipy-trf"]:
+        eq, result = eq.solve(
+            maxiter=3,
+            constraints=constraints,
+            objective=obj,
+            optimizer=opt,
+            ftol=1e-12,
+            xtol=1e-12,
+            gtol=1e-12,
+        )
+        assert "Maximum number of iterations has been exceeded" in result["message"]
+    objectives = Energy()
+    obj = ObjectiveFunction(objectives)
+    for opt in ["scipy-trust-exact"]:
+        eq, result = eq.solve(
+            maxiter=3,
+            constraints=constraints,
+            objective=obj,
+            optimizer=opt,
+            ftol=1e-12,
+            xtol=1e-12,
+            gtol=1e-12,
+        )
+        # result["message"] is a list for scipy-trust-exact but a string for scipy-trf
+        # why?
+        assert "Maximum number of iterations has been exceeded" in result["message"][0]
```

Comment:
Figure this out before finishing this PR. We should have a consistent `result["message"]` type across all optimizers.